tests/lib/state_test.py

```diff
@@ -132,6 +132,22 @@ def test_sort_multiple_ip(self):
 
         desired_state.verify_interfaces(current_state)
 
+    def test_description_is_empty(self):
+        desired_state = self._base_state
+        current_state = self._base_state
+        desired_state.interfaces['foo-name'][Interface.DESCRIPTION] = ''
+
+        desired_state.verify_interfaces(current_state)
+
+    def test_description_is_not_empty(self):
+        desired_state = self._base_state
+        current_state = self._base_state
+        desired_state.interfaces['foo-name'][Interface.DESCRIPTION] = 'bar'
+        current_state.interfaces['foo-name'][Interface.DESCRIPTION] = 'bar'
+
+        desired_state.verify_interfaces(current_state)
+        desired_state.interfaces['foo-name'][Interface.DESCRIPTION]
```

Comment:
What is this line all about?

Reply:
> In the negative case test, we are checking that we have not removed the field "description" as it is not empty. The last line causes an exception if the field "description" does not exist.

In python, this is causing the function/method to return the value.
But in general, it looks like a hack/abuse of the language options.

> 
> Another option could be to use an assert statement.

If you want to check if a key exists, check it with an assert please. `assert k in d`.

But.. checking if the description key exists or not is an implementation detail which the test should not care about.
It is an assumption that `verify_interfaces` is mutable, it could have been immutable (or changed to be so in the future).

Reply:
Seems fine, I am going to remove that line. Thanks!